File: Day_22/Q85.cpp
#include <bits/stdc++.h>
using namespace std;

//Check Palindrome
bool palindrome(string str){
    int start = 0;
    int end = str.length() - 1;

    while(start < end){
        if(str[start] != str[end]) return false;

        start++;
        end--;
    }

    return true;
}

int main(){
    string str;
    cin >> str;

    if(palindrome(str)) cout << "Palindrome";
    else cout << "Not Palindrome";

    return 0;
}

File: Day_22/Q85.java
import java.util.*;

class Q85{
    static Scanner sc = new Scanner(System.in);

    //Check Palindrome
    static boolean palindrome(String str){
        int start = 0;
        int end = str.length() - 1;

        while(start < end){
            if(str.charAt(start) != str.charAt(end)) return false;

            start++;
            end--;
        }

        return true;
    }

    public static void main(String args[]){
        String str = sc.next();

        if(palindrome(str)) System.out.println("Palindrome");
        else System.out.println("Not Palindrome");

        sc.close();
    }
}
